extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -471,7 +464,7 @@ func (a *genericActuator) waitUntilUnwantedMachineDeploymentsDeleted(ctx context
 	})
 }
 
-func (a *genericActuator) updateWorkerStatusMachineDeployments(ctx context.Context, worker *extensionsv1alpha1.Worker, machineDeployments extensionsworker.MachineDeployments, isRollingUpdate bool) error {
+func (a *genericActuator) updateWorkerStatusMachineDeployments(ctx context.Context, worker *extensionsv1alpha1.Worker, machineDeployments extensionsworker.MachineDeployments) error {
 	var statusMachineDeployments []extensionsv1alpha1.MachineDeployment
 
 	for _, machineDeployment := range machineDeployments {
```

Comment:
If the machineDeployments array are empty I guess you could skip the entire function by returning nil. This means that you will not need the `if len(statusMachineDeployments) > 0 {` check on line 479

Reply:
Nice suggestion, we added early exit logic as suggested.